packages/cli/src/commands/install.ts

```diff
@@ -0,0 +1,125 @@
+
+export type InstallArgs = {
+  path?: string;
+  silent?: boolean;
+};
+
+const defaultPath = path.join(os.homedir(), 'logto');
+const pgRequired = new semver.SemVer('14.0.0');
+
+const validateNodeVersion = () => {
+  const required = new semver.SemVer('16.0.0');
+  const current = new semver.SemVer(execSync('node -v', { encoding: 'utf8', stdio: 'pipe' }));
+
+  if (required.compare(current) > 0) {
+    log.error(`Logto requires NodeJS >=${required.version}, but ${current.version} found.`);
+  }
+
+  if (current.major > required.major) {
+    log.warn(
+      `Logto is tested under NodeJS ^${required.version}, but version ${current.version} found.`
+    );
+  }
+};
+
+const validatePath = (value: string) =>
+  existsSync(path.resolve(value))
+    ? `The path ${chalk.green(value)} already exists, please try another.`
+    : true;
+
+const getInstancePath = async () => {
+  const { hasPostgresUrl } = await inquirer.prompt<{ hasPostgresUrl?: boolean }>({
+    name: 'hasPostgresUrl',
+    message: `Logto requires PostgreSQL >=${pgRequired.version} but cannot find in the current environment.\n  Do you have a remote PostgreSQL instance ready?`,
+    type: 'confirm',
+    when: () => {
+      const pgOutput = safeExecSync('postgres --version') ?? '';
+      // Filter out all brackets in the output since Homebrew will append `(Homebrew)`.
+      const pgArray = pgOutput.split(' ').filter((value) => !value.startsWith('('));
+      const pgCurrent = semver.coerce(pgArray[pgArray.length - 1]);
+
+      return !pgCurrent || pgCurrent.compare(pgRequired) < 0;
+    },
+  });
+
+  if (hasPostgresUrl === false) {
+    log.error('Logto requires a Postgres instance to run.');
+  }
+
+  const { instancePath } = await inquirer.prompt<{ instancePath: string }>({
+    name: 'instancePath',
+    message: 'Where should we create your Logto instance?',
+    type: 'input',
+    default: defaultPath,
+    filter: (value: string) => value.trim(),
+    validate: validatePath,
+  });
+
+  return instancePath;
+};
+
+const downloadRelease = async () => {
+  const tarFilePath = path.resolve(os.tmpdir(), './logto.tar.gz');
+
+  log.info(`Download Logto to ${tarFilePath}`);
+  await downloadFile(
+    'https://github.com/logto-io/logto/releases/latest/download/logto.tar.gz',
+    tarFilePath
+  );
+
+  return tarFilePath;
+};
+
+const decompress = async (toPath: string, tarPath: string) => {
+  const decompressSpinner = ora({
+    text: `Decompress to ${toPath}`,
+    prefixText: chalk.blue('[info]'),
+  }).start();
+
+  try {
+    await mkdir(toPath);
+    await tar.extract({ file: tarPath, cwd: toPath, strip: 1 });
+  } catch (error: unknown) {
+    decompressSpinner.fail();
+    log.error(error);
+
+    return;
+  }
+
+  decompressSpinner.succeed();
+};
+
+const install = async ({ path: pathArgument = defaultPath, silent = false }: InstallArgs) => {
+  validateNodeVersion();
+
+  const instancePath = (!silent && (await getInstancePath())) || pathArgument;
```

Comment:
if !silent. but with provided pathArgument. should we jump the getInstancePath, or maybe pass the pathArgument into the getInstancePath method, so it can be picked up automatically?

Reply:
good call, i did it in [this PR](https://github.com/logto-io/logto/pull/2051/files#diff-f8813d708357a4cb2374550bf633d99515fb486095c8f69937255795bbda5007R42)